### ISSUE TITLE
Panic: gosec: package "conversion": runtime error: index out of range [2] with length 2

### ISSUE THREAD

**Issue by drew-richardson** [golangci/golangci-lint#3071]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

golangci-lint gosec crashes on
```go
new(big.Rat).SetString("")
```

This issue does not happen when using the [prebuild gosec](https://github.com/securego/gosec/releases/tag/v2.12.0) binaries, but does happen when built from source. I did report the issue to [gosec](https://github.com/securego/gosec/issues/843) but am concerned they'll just tell me I'm building gosec wrong so I'm also reporting here.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.48.0 built from 2d8fea81 on 2022-08-04T18:44:38Z
Smart, fast linters runner. Run it in cloud for every GitHub pull request on https://golangci.com

Usage:
  golangci-lint [flags]
  golangci-lint [command]

Available Commands:
  cache       Cache control and information
  completion  Generate the autocompletion script for the specified shell
  config      Config
  help        Help
  linters     List current linters configuration
  run         Run the linters
  version     Version

Flags:
      --color string              Use color when printing; can be 'always', 'auto', or 'never' (default "auto")
  -j, --concurrency int           Concurrency (default NumCPU) (default 12)
      --cpu-profile-path string   Path to CPU profile output file
  -h, --help                      help for golangci-lint
      --mem-profile-path string   Path to memory profile output file
      --trace-path string         Path to trace output file
  -v, --verbose                   verbose output
      --version                   Print version

Use "golangci-lint [command] --help" for more information about a command.
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
linters:
  enable:
    - gosec
$ 
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/drichardson/Library/Caches/go-build"
GOENV="/Users/drichardson/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/drichardson/go/pkg/mod"
GONOPROXY="gitlab.frg.tech,jaxf-github.fanatics.corp"
GONOSUMDB="gitlab.frg.tech,jaxf-github.fanatics.corp"
GOOS="darwin"
GOPATH="/Users/drichardson/go"
GOPRIVATE="gitlab.frg.tech,jaxf-github.fanatics.corp"
GOPROXY="direct"
GOROOT="/usr/local/go"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GOVCS=""
GOVERSION="go1.19"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/tmp/conversion/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/_4/wfyg_8f57j3_1pm7qzbkk2z80000gn/T/go-build3571043214=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /tmp/conversion /tmp / /Users/drichardson] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosec gosimple govet ineffassign staticcheck typecheck unused varcheck] 
INFO [loader] Go packages loading at mode 575 (deps|name|types_sizes|compiled_files|files|imports|exports_file) took 142.879657ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 169.621µs 
INFO [linters context/goanalysis] analyzers took 35.061448478s with top 10 stages: buildir: 9.228494935s, ctrlflow: 8.320643479s, fact_deprecated: 8.300431435s, printf: 8.286000462s, SA5012: 288.096503ms, fact_purity: 206.625295ms, nilness: 195.667876ms, typedness: 185.970895ms, inspect: 45.534144ms, ineffassign: 363.388µs 
ERRO [runner] Panic: gosec: package "conversion" (isInitialPkg: true, needAnalyzeSource: true): runtime error: index out of range [2] with length 2: goroutine 1256 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func1()
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:101 +0x155
panic({0x1c206c0, 0xc0047ca108})
	runtime/panic.go:884 +0x212
github.com/securego/gosec/v2.GoVersion()
	github.com/securego/gosec/v2@v2.12.0/helpers.go:458 +0xee
github.com/securego/gosec/v2/rules.(*usingOldMathBig).Match(0xc0004c46c0, {0x1ee5800, 0xc0004c4240}, 0x4?)
	github.com/securego/gosec/v2@v2.12.0/rules/math_big_rat.go:24 +0x45
github.com/securego/gosec/v2.(*Analyzer).Visit(0xc001c36000, {0x1ee5800, 0xc0004c4240})
	github.com/securego/gosec/v2@v2.12.0/analyzer.go:477 +0x843
go/ast.Walk({0x1ee03e0?, 0xc001c36000?}, {0x1ee5800?, 0xc0004c4240?})
	go/ast/walk.go:51 +0x62
go/ast.Walk({0x1ee03e0?, 0xc001c36000?}, {0x1ee5a30?, 0xc00a0ca300?})
	go/ast/walk.go:206 +0xa10
go/ast.walkStmtList({0x1ee03e0, 0xc001c36000}, {0xc00a0ca310?, 0x1, 0x1bb1100?})
	go/ast/walk.go:32 +0x91
go/ast.Walk({0x1ee03e0?, 0xc001c36000?}, {0x1ee57b0?, 0xc000b252c0?})
	go/ast/walk.go:234 +0x606
go/ast.Walk({0x1ee03e0?, 0xc001c36000?}, {0x1ee5af8?, 0xc000b25560?})
	go/ast/walk.go:357 +0x165c
go/ast.walkDeclList({0x1ee03e0, 0xc001c36000}, {0xc0092ceda0?, 0x2, 0x110?})
	go/ast/walk.go:38 +0x91
go/ast.Walk({0x1ee03e0?, 0xc001c36000?}, {0x1ee5aa8?, 0xc00868c000?})
	go/ast/walk.go:366 +0xdea
github.com/securego/gosec/v2.(*Analyzer).Check(0xc001c36000, 0xc00045b9a0)
	github.com/securego/gosec/v2@v2.12.0/analyzer.go:301 +0x545
github.com/golangci/golangci-lint/pkg/golinters.runGoSec(0xc0008a0230, 0xc00b74cc30, 0xc0004ee5a8, 0xc001c36000)
	github.com/golangci/golangci-lint/pkg/golinters/gosec.go:88 +0xcc
github.com/golangci/golangci-lint/pkg/golinters.NewGosec.func1.1(0xc00b74cc30?)
	github.com/golangci/golangci-lint/pkg/golinters/gosec.go:67 +0x22a
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyze(0xc00115e120)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:187 +0x9d6
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe.func2()
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:105 +0x1d
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0xc000fa0f00, {0x1c8115f, 0x5}, 0xc000b45f48)
	github.com/golangci/golangci-lint/pkg/timeutils/stopwatch.go:111 +0x4a
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*action).analyzeSafe(0xc000049bc0?)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_action.go:104 +0x85
github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze.func2(0xc00115e120)
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:80 +0xb4
created by github.com/golangci/golangci-lint/pkg/golinters/goanalysis.(*loadingPackage).analyze
	github.com/golangci/golangci-lint/pkg/golinters/goanalysis/runner_loadingpackage.go:75 +0x1eb 
WARN [runner] Can't run linter goanalysis_metalinter: goanalysis_metalinter: gosec: package "conversion" (isInitialPkg: true, needAnalyzeSource: true): runtime error: index out of range [2] with length 2 
INFO [runner] processing took 7.482µs with stages: max_same_issues: 2.485µs, nolint: 2.416µs, skip_dirs: 415ns, max_from_linter: 261ns, cgo: 192ns, filename_unadjuster: 163ns, exclude: 160ns, path_prettifier: 159ns, autogenerated_exclude: 157ns, skip_files: 152ns, uniq_by_line: 151ns, source_code: 102ns, max_per_file_from_linter: 91ns, identifier_marker: 88ns, severity-rules: 86ns, diff: 83ns, sort_results: 83ns, path_shortener: 82ns, exclude-rules: 80ns, path_prefixer: 76ns 
INFO [runner] linters took 3.330602173s with stages: goanalysis_metalinter: 3.330546389s 
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: goanalysis_metalinter: gosec: package "conversion" (isInitialPkg: true, needAnalyzeSource: true): runtime error: index out of range [2] with length 2
 
INFO Memory: 36 samples, avg is 196.1MB, max is 271.5MB 
INFO Execution took 3.487093875s                  
```

</details>

### Code example or link to a public repository

<details>

go.mod:
```
module conversion

go 1.19
```

conversion.go:
```go
package conversion

import "math/big"

func Conversion() {
	new(big.Rat).SetString("")
}
```

</details>

**Comment by ldez** [comment]
Hello,

The problem is related to an incorrect assumption about the format of the Go version:
https://github.com/securego/gosec/blob/0c8e63ed86f8fc2da42787e0a653114490078d6c/helpers.go#L476-L482

I created a fix https://github.com/securego/gosec/pull/844
